src/Persistence/MassTransit.EntityFrameworkCoreIntegration/EntityFrameworkCoreIntegration/EntityFrameworkOutboxContextFactory.cs

```diff
@@ -44,10 +45,10 @@ async Task<bool> Execute()
 
                 try
                 {
-                    var inboxState = await _dbContext.Set<InboxState>()
+                    var inboxStateList = await _dbContext.Set<InboxState>()
                         .FromSqlRaw(_lockStatement, messageId, options.ConsumerId)
-                        .AsTracking()
-                        .SingleOrDefaultAsync(context.CancellationToken).ConfigureAwait(false);
+                        .AsTracking().ToListAsync(context.CancellationToken).ConfigureAwait(false);
```

Comment:
Again, why?

Reply:
Same as above.

Reply:
I made it same when I changed the above code. However here the subquery is still generated
The statement pushed do dbms is: 

```
SELECT "s"."CorrelationId", "s"."CurrentState", "s"."EventId", "s"."MemberId", "s"."Payment", "s"."RegistrationDate"
FROM (
    SELECT * FROM "RegistrationState" WHERE "CorrelationId" = :p0
) "s"
FETCH FIRST 2 ROWS ONLY

```

I think the parameters are the problem, when there are parameters used, the subquery is used and in this case


When the commented code in OracleLockStatementFormatter.Complete() has the " FOR UPDATE" it will fail with ORA-00907: Missing right parenthesis,  that is why the " FOR UPDATE" is commented out.
It looks like the FOR UPDATE cannot be used in sub query in Oracle.